docs/sdk/src/reference_docs/development_environment_advice.rs

```diff
@@ -0,0 +1,95 @@
+//! # Development Environment Advice
+//!
+//! Large Rust projects are known for sometimes long compile times and sluggish dev tooling, and
+//! polkadot-sdk is no exception.
+//!
+//! This page contains some advice to improve your workflow when using common tooling.
+//!
+//! ## Rust Analyzer Configuration
+//!
+//! [Rust Analyzer](https://rust-analyzer.github.io/) is the defacto [LSP](https://langserver.org/) for Rust. Its default
+//! settings are fine for smaller projects, but not well optimised for polkadot-sdk.
+//!
+//! Below is a suggested configuration in Lua (can use with `neovim/nvim-lspconfig`)
+//! with comments describing the rationale for each setting:
+//!
+//! ```lua
+//! ["rust-analyzer"] = {
+//!   cargo = {
+//!     # Check feature-gated code
+//!     features = "all",
+//!     extraEnv = {
+//!       # Use a seperate target dir for Rust Analyzer. Helpful if you want to use Rust
+//!       # Analyzer and cargo on the command line at the same time.
+//!       ["CARGO_TARGET_DIR"] = "target/nvim-rust-analyzer",
+//!       # Skip building WASM, there is never need for it here
+//!       ["SKIP_WASM_BUILD"] = "1",
+//!       # Improve stability
+//!       ["CHALK_OVERFLOW_DEPTH"] = "100000000",
+//!     },
+//!   },
+//!   procMacro = {
+//!     # Don't expand some problematic proc_macros
+//!     ignored = {
+//!       ["async-trait"] = { "async_trait" },
+//!       ["napi-derive"] = { "napi" },
+//!       ["async-recursion"] = { "async_recursion" },
+//!       ["async-std"] = { "async_std" },
+//!     },
+//!   },
+//!   rustfmt = {
+//!     # Use nightly formatting.
+//!     # See the polkadot-sdk CI job that checks formatting for the current version used in
+//!     # polkadot-sdk.
+//!     extraArgs = { "+nightly-2023-11-01" },
+//!   },
+//! },
+//! ```
+//!
+//! and the same configuration in JSON for VSCode's `settings.json`:
+//!
+//! ```json
+//! {
+//!   "rust-analyzer.cargo.features": "all",
+//!   "rust-analyzer.cargo.extraEnv": {
```

Comment:
I don't know how to verify if `SKIP_WASM_BUILD` and `CHALK_OVERFLOW_DEPTH` are set but for `CARGO_TARGET_DIR` I do need to set it in `rust-analyzer.server.extraEnv` instead of `rust-analyzer.cargo.extraEnv`.

Reply:
Hmm strange, I checked again and `rust-analyzer.cargo.extraEnv` works for creating a dedicated dir under `target/`.

What version of VSCode and the rust-analyzer plugin are you using?

Reply:
I just found that `rust-analyzer.rust.analyzerTargetDir` was added as an option recently, and is now the idiomatic way to specify a target dir override. Will update options to use that.

Reply:
Is `rust-analyzer.rust.analyzerTargetDir` working for you?